--- a/sources/pyside6/PySide6/QtNetwork/CMakeLists.txt
+++ b/sources/pyside6/PySide6/QtNetwork/CMakeLists.txt
@@ -9,48 +9,29 @@
 ${QtNetwork_GEN_DIR}/qabstractnetworkcache_wrapper.cpp
 ${QtNetwork_GEN_DIR}/qabstractsocket_wrapper.cpp
 ${QtNetwork_GEN_DIR}/qauthenticator_wrapper.cpp
-${QtNetwork_GEN_DIR}/qdnsdomainnamerecord_wrapper.cpp
-${QtNetwork_GEN_DIR}/qdnshostaddressrecord_wrapper.cpp
-${QtNetwork_GEN_DIR}/qdnslookup_wrapper.cpp
-${QtNetwork_GEN_DIR}/qdnsmailexchangerecord_wrapper.cpp
-${QtNetwork_GEN_DIR}/qdnsservicerecord_wrapper.cpp
-${QtNetwork_GEN_DIR}/qdnstextrecord_wrapper.cpp
-${QtNetwork_GEN_DIR}/qdnstlsassociationrecord_wrapper.cpp
-${QtNetwork_GEN_DIR}/qformdatabuilder_wrapper.cpp
-${QtNetwork_GEN_DIR}/qformdatapartbuilder_wrapper.cpp
 ${QtNetwork_GEN_DIR}/qhostaddress_wrapper.cpp
 ${QtNetwork_GEN_DIR}/qhostinfo_wrapper.cpp
 ${QtNetwork_GEN_DIR}/qhstspolicy_wrapper.cpp
-${QtNetwork_GEN_DIR}/qhttp1configuration_wrapper.cpp
 ${QtNetwork_GEN_DIR}/qhttpheaders_wrapper.cpp
-${QtNetwork_GEN_DIR}/qhttpmultipart_wrapper.cpp
-${QtNetwork_GEN_DIR}/qhttppart_wrapper.cpp
-${QtNetwork_GEN_DIR}/qhttp2configuration_wrapper.cpp
 ${QtNetwork_GEN_DIR}/qipv6address_wrapper.cpp
 ${QtNetwork_GEN_DIR}/qlocalserver_wrapper.cpp
 ${QtNetwork_GEN_DIR}/qlocalsocket_wrapper.cpp
 ${QtNetwork_GEN_DIR}/qnetworkaccessmanager_wrapper.cpp
-${QtNetwork_GEN_DIR}/qnetworkaddressentry_wrapper.cpp
 ${QtNetwork_GEN_DIR}/qnetworkcachemetadata_wrapper.cpp
 ${QtNetwork_GEN_DIR}/qnetworkcookie_wrapper.cpp
 ${QtNetwork_GEN_DIR}/qnetworkcookiejar_wrapper.cpp
 ${QtNetwork_GEN_DIR}/qnetworkdatagram_wrapper.cpp
 ${QtNetwork_GEN_DIR}/qnetworkdiskcache_wrapper.cpp
 ${QtNetwork_GEN_DIR}/qnetworkinformation_wrapper.cpp
-${QtNetwork_GEN_DIR}/qnetworkinterface_wrapper.cpp
 ${QtNetwork_GEN_DIR}/qnetworkproxy_wrapper.cpp
 ${QtNetwork_GEN_DIR}/qnetworkproxyfactory_wrapper.cpp
 ${QtNetwork_GEN_DIR}/qnetworkproxyquery_wrapper.cpp
 ${QtNetwork_GEN_DIR}/qnetworkreply_wrapper.cpp
 ${QtNetwork_GEN_DIR}/qnetworkrequest_wrapper.cpp
-${QtNetwork_GEN_DIR}/qnetworkrequestfactory_wrapper.cpp
 ${QtNetwork_GEN_DIR}/qpassworddigestor_wrapper.cpp
-${QtNetwork_GEN_DIR}/qrestaccessmanager_wrapper.cpp
-${QtNetwork_GEN_DIR}/qrestreply_wrapper.cpp
 ${QtNetwork_GEN_DIR}/qssl_wrapper.cpp
 ${QtNetwork_GEN_DIR}/qsslcertificate_wrapper.cpp
 ${QtNetwork_GEN_DIR}/qsslcertificateextension_wrapper.cpp
-${QtNetwork_GEN_DIR}/qsslellipticcurve_wrapper.cpp
 ${QtNetwork_GEN_DIR}/qtcpserver_wrapper.cpp
 ${QtNetwork_GEN_DIR}/qtcpsocket_wrapper.cpp
 ${QtNetwork_GEN_DIR}/qudpsocket_wrapper.cpp
@@ -64,7 +45,8 @@
 if("ssl" IN_LIST QtNetwork_disabled_features)
     list(APPEND QtNetwork_DROPPED_ENTRIES QOcspResponse QSslCipher
          QSslConfiguration QSslDiffieHellmanParameters QSslError
-         QSslKey QSslPreSharedKeyAuthenticator QSslSocket QSslServer)
+         QSslKey QSslPreSharedKeyAuthenticator QSslSocket QSslServer
+         QSslEllipticCurve)
     message(STATUS "Qt${QT_MAJOR_VERSION}Network: Dropping SSL classes")
 else()
     # Problems with operator==(QSslEllipticCurve,QSslEllipticCurve)
@@ -78,7 +60,8 @@
          ${QtNetwork_GEN_DIR}/qsslpresharedkeyauthenticator_wrapper.cpp
          ${QtNetwork_GEN_DIR}/qsslserver_wrapper.cpp
          ${QtNetwork_GEN_DIR}/qsslsocket_wrapper.cpp
-         ${QtNetwork_GEN_DIR}/qocspresponse_wrapper.cpp)
+         ${QtNetwork_GEN_DIR}/qocspresponse_wrapper.cpp
+         ${QtNetwork_GEN_DIR}/qsslellipticcurve_wrapper.cpp)
     message(STATUS "Qt${QT_MAJOR_VERSION}Network: Adding SSL classes")
 endif()
 
@@ -103,6 +86,50 @@
     message(STATUS "Qt${QT_MAJOR_VERSION}Network: Adding SCTP classes")
 endif()
 
+if("http" IN_LIST QtNetwork_disabled_features)
+    list(APPEND QtNetwork_DROPPED_ENTRIES QHttpMultiPart QHttpPart QHttp1Configuration QHttp2Configuration
+         QFormDataBuilder QFormDataPartBuilder QNetworkRequestFactory QRestAccessManager QRestReply)
+    message(STATUS "Qt${QT_MAJOR_VERSION}Network: Dropping http classes")
+else()
+    list(APPEND QtNetwork_SRC
+         ${QtNetwork_GEN_DIR}/qformdatabuilder_wrapper.cpp
+         ${QtNetwork_GEN_DIR}/qformdatapartbuilder_wrapper.cpp
+         ${QtNetwork_GEN_DIR}/qhttp1configuration_wrapper.cpp
+         ${QtNetwork_GEN_DIR}/qhttpmultipart_wrapper.cpp
+         ${QtNetwork_GEN_DIR}/qhttppart_wrapper.cpp
+         ${QtNetwork_GEN_DIR}/qhttp2configuration_wrapper.cpp
+         ${QtNetwork_GEN_DIR}/qnetworkrequestfactory_wrapper.cpp
+         ${QtNetwork_GEN_DIR}/qrestaccessmanager_wrapper.cpp
+         ${QtNetwork_GEN_DIR}/qrestreply_wrapper.cpp)
+    message(STATUS "Qt${QT_MAJOR_VERSION}Network: Adding http classes")
+endif()
+
+if("dnslookup" IN_LIST QtNetwork_disabled_features)
+    list(APPEND QtNetwork_DROPPED_ENTRIES QDnsDomainNameRecord QDnsHostAddressRecord QDnsLookup
+         QDnsMailExchangeRecord QDnsServiceRecord QDnsTextRecord QDnsTlsAssociationRecord)
+    message(STATUS "Qt${QT_MAJOR_VERSION}Network: Dropping dnslookup classes")
+else()
+    list(APPEND QtNetwork_SRC
+         ${QtNetwork_GEN_DIR}/qdnsdomainnamerecord_wrapper.cpp
+         ${QtNetwork_GEN_DIR}/qdnshostaddressrecord_wrapper.cpp
+         ${QtNetwork_GEN_DIR}/qdnslookup_wrapper.cpp
+         ${QtNetwork_GEN_DIR}/qdnsmailexchangerecord_wrapper.cpp
+         ${QtNetwork_GEN_DIR}/qdnsservicerecord_wrapper.cpp
+         ${QtNetwork_GEN_DIR}/qdnstextrecord_wrapper.cpp
+         ${QtNetwork_GEN_DIR}/qdnstlsassociationrecord_wrapper.cpp)
+    message(STATUS "Qt${QT_MAJOR_VERSION}Network: Adding dnslookup classes")
+endif()
+
+if("networkinterface" IN_LIST QtNetwork_disabled_features)
+    list(APPEND QtNetwork_DROPPED_ENTRIES QNetworkInterface QNetworkAddressEntry)
+    message(STATUS "Qt${QT_MAJOR_VERSION}Network: Dropping networkinterface classes")
+else()
+    list(APPEND QtNetwork_SRC
+         ${QtNetwork_GEN_DIR}/qnetworkaddressentry_wrapper.cpp
+         ${QtNetwork_GEN_DIR}/qnetworkinterface_wrapper.cpp)
+    message(STATUS "Qt${QT_MAJOR_VERSION}Network: Adding networkinterface classes")
+endif()
+
 set(QtNetwork_include_dirs  ${QtNetwork_SOURCE_DIR}
                             ${QtNetwork_BINARY_DIR}
                             ${Qt${QT_MAJOR_VERSION}Core_INCLUDE_DIRS}
